File: model/dtos/hospital.py
from pydantic import BaseModel, EmailStr
from typing import Optional, List
from datetime import datetime

class HospitalBase(BaseModel):
    name: str
    address: str
    capacity: int

class HospitalDTO(HospitalBase):
    id: int
    created_at: datetime
    updated_at: datetime
    is_active: bool = True

    class Config:
        from_attributes = True

class HospitalCreateDTO(HospitalBase):
    pass

class HospitalResponseDTO(HospitalBase):
    id: int

    class Config:
        orm_mode = True

File: frontend/src/types/hospital.ts
export interface HospitalBaseDTO {
    name: string;
    address: string;
    capacity: number;
}

export interface HospitalCreateDTO extends HospitalBaseDTO {}

export interface HospitalUpdateDTO {
    name?: string;
    address?: string;
    capacity?: number;
}

export interface Hospital {
    id: number;
    name: string;
    address: string;
    capacity: number;
    created_at: string;
    updated_at: string;
    is_active: boolean;
} 